packages/orderbook/src/orderbook.ts

```diff
@@ -217,28 +228,95 @@ export class Orderbook {
       };
     }
 
+    // not fool-proof but scenarios where smart contract wallet is not deployed will be an edge case
+    const isSmartContractWallet: boolean = await this.orderbookConfig.provider.getCode(makerAddress) !== '0x';
+    if (isSmartContractWallet) {
+      track('orderbookmr', 'bulkListings', { walletType: 'Passport', makerAddress, listingsCount: listingParams.length });
+
+      // eslint-disable-next-line max-len
+      const prepareListingResponses = await Promise.all(listingParams.map((listing) => this.seaport.prepareSeaportOrder(
+        makerAddress,
+        listing.sell,
+        listing.buy,
+        new Date(),
+        listing.orderExpiry || Orderbook.defaultOrderExpiry(),
+      )));
+
+      const pendingApproval: string[] = [];
+      const actions = prepareListingResponses.flatMap((response) => {
+        // de-dupe approval transactions to ensure every contract has
+        // a maximum of 1 approval transaction
+        const dedupedActions: Action[] = [];
+        response.actions.forEach((action) => {
```

Comment:
idiomatically I think this would be better as a `for (const action of response.actions)` as its side-effecting

Reply:
@Sam-Jeston eslint doesn't like updating the loop and shouts with,
`ESLint: iterators/generators require regenerator-runtime, which is too heavyweight for this guide to allow them. Separately, loops should be avoided in favor of array iterations.(no-restricted-syntax)`